src/controllers/waitForOthersToSign.controller.ts

```diff
@@ -2,33 +2,44 @@ import { inject } from 'inversify'
 import { controller, httpGet } from 'inversify-express-utils'
 
 import BaseController from 'app/controllers/base.controller'
+import DissolutionGetDirector from 'app/models/dto/dissolutionGetDirector'
+import DissolutionGetResponse from 'app/models/dto/dissolutionGetResponse'
 import OfficerType from 'app/models/dto/officerType.enum'
+import Optional from 'app/models/optional'
 import DissolutionSession from 'app/models/session/dissolutionSession.model'
 import { WAIT_FOR_OTHERS_TO_SIGN_URI } from 'app/paths'
+import DissolutionService from 'app/services/dissolution/dissolution.service'
 import SessionService from 'app/services/session/session.service'
 
 interface ViewModel {
   officerType: OfficerType
+  signatories: DissolutionGetDirector[]
 }
 
 @controller(WAIT_FOR_OTHERS_TO_SIGN_URI)
 export class WaitForOthersToSignController extends BaseController {
 
   public constructor(
-    @inject(SessionService) private session: SessionService) {
+    @inject(SessionService) private session: SessionService,
+    @inject(DissolutionService) private dissolutionService: DissolutionService
+  ) {
     super()
   }
 
   @httpGet('')
   public async get(): Promise<string> {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
     const session: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
 
-    return this.renderView(session.officerType!)
+    const dissolution: Optional<DissolutionGetResponse> = await this.dissolutionService.getDissolution(token, session)
```

Comment:
Just a thought, what if you had:
```
const applicationStatus: ApplicationStatusViewModel = await this.applicationStatusService.generateApplicationStatusViewModel(token, session)
```
where `ApplicationStatusViewModel` is the object used for the table / rows in nunjucks, keeping the template as simple as possible.

Reply:
Yep. Could do the same change for the Payment Review page

Reply:
As noted in another comment, some logic has been moved out to a mapper/service to reduce controller and template complexity